packages/wonder-blocks-dropdown/components/dropdown-launcher.js

```diff
@@ -0,0 +1,244 @@
+        const {selectedValues} = this.props;
+        return React.Children.toArray(this.props.menuItems)
+            .filter(Boolean)
+            .map((item) => {
+                const {value, disabled} = item.props;
+                const itemObject = {
+                    component: item,
+                    focusable:
+                        ActionItem.isClassOf(item) || OptionItem.isClassOf(item)
+                            ? !disabled
+                            : false,
+                    populatedProps: {},
+                };
+                if (ActionItem.isClassOf(item)) {
+                    return {
+                        ...itemObject,
+                        populatedProps: {
+                            indent: true,
+                            onClick: this.handleItemSelected,
+                        },
+                    };
+                } else if (OptionItem.isClassOf(item)) {
+                    return {
+                        ...itemObject,
+                        populatedProps: {
+                            onToggle: this.handleOptionSelected,
+                            selected: selectedValues
+                                ? selectedValues.includes(value)
+                                : false,
+                            variant: "check",
+                        },
+                    };
+                } else {
+                    return itemObject;
+                }
+            });
+    };
+
+    handleOpenerRef = (node: any) => {
+        this.openerElement = ((ReactDOM.findDOMNode(node): any): HTMLElement);
+    };
+
+    render() {
+        const dropdownItems = this._getMenuItems();
+        const {
+            children,
+            alignment,
+            disabled,
+            style,
+            dropdownStyle,
+            /* eslint-disable no-unused-vars */
+            onChange,
+            selectedValues,
+            "aria-disabled": ariaDisabled,
+            ...sharedProps
+        } = this.props;
+
+        const opener = (
+            <span
+                ref={this.handleOpenerRef}
+                disabled={!dropdownItems.length || disabled}
+            >
+                {children(this._openDropdown)}
+            </span>
+        );
+
+        return (
+            <Dropdown
+                alignment={alignment}
+                items={this._getMenuItems()}
+                keyboard={this.state.keyboard}
+                dropdownStyle={[styles.menuTopSpace, dropdownStyle]}
+                onOpenChanged={this.handleOpenChanged}
+                open={this.state.open}
+                opener={opener}
+                openerElement={this.openerElement}
+                role="menu"
+                style={style}
+            />
+        );
+    }
+}
+
+const styles = StyleSheet.create({
+    caret: {
+        marginLeft: 4,
+    },
+
+    // The design calls for additional offset around the opener.
+    opener: {
```

Comment:
Not sure if you are using this style inside the component, could you please confirm?

Reply:
its not being used, removing it makes no difference

Reply:
Even though this is a prototype removing extraneous details will help people understand what the prototype more easily.